metrics/prometheus-metrics-endpoints-and-auto-discovery.md

```diff
@@ -106,6 +106,27 @@ datasources = /etc/grafana-datasources
 
 *NOTE:* This approach for making default dashboards available will be used until such a time as a more automated discovery of dashboards is determined. For example, is it possible to store the dashboard defininition with the Service that has metrics available e.g. Keycloak, and then make it show up in Grafana.
 
+#### Auto Discovery of Service Dashboard Definitions
+
+An approach to automated discovery of dashboard definitions is being investigated. The initial discussion of this approach can be seen [here](https://github.com/aerogear/proposals/pull/5#discussion_r157767612).
+
+The approach involves a sidecar service running inside the same pod as Grafana. This service would use the Kubernetes API to watch the namespace for Config Maps containing a Grafana dashboard definition (JSON). These Config Maps will be identified with an annotation. For example:
+
+```
+org.aerogear.grafana.dashboard: true
+```
+
+The sidecar service would feed the disovered dashboard JSON definitions into Grafana using an API call.
```

Comment:
I think someone already mentioned this but it would be nice if we could show a Notification (in Openshifts Notification tab) whenever a new dashboard is imported/updated. Possibly by creating an event.

Reply:
@pb82 I was wrong. 

Notifications are just filtered Api events: https://github.com/spadgett/origin-web-console/blob/57d0abd747de201e6e757a170f910e58c2c03ded/app/scripts/directives/notifications/notificationDrawerWrapper.js#L159

This will require us to extend events, which is not trivial task.